convex/lib/honoWithConvex.ts

```diff
@@ -1,15 +1,56 @@
-import { HttpRouter, PublicHttpEndpoint, RoutableMethod } from "convex/server";
+import {
+  HttpRouter,
+  PublicHttpEndpoint,
+  RoutableMethod,
+  ROUTABLE_HTTP_METHODS,
+} from "convex/server";
 import { Hono } from "hono";
 import { httpEndpoint, HttpEndpointCtx } from "../_generated/server";
 
-// Idk where this type is supposed to come from but hack to get it working
+/**
+ * Hono uses the `FetchEvent` type internally, which has to do with service workers
+ * and isn't included in the Convex tsconfig.
+ *
+ * As a workaround, define this type here so Hono + Convex compiles.
```

Comment:
Some googling indicates that I could add `webworker` to my `tsconfig.json` under `lib`, but this is incompatible with `dom` (which is what our default `convex/tsconfig.json` uses), so sticking with this hack but with a better comment